src/fparser/tests/test_parser.py

```diff
@@ -647,6 +670,146 @@ def test_integer():
+    assert parse(DoublePrecision, 'double precision*4') == 'DOUBLEPRECISION*4'
+    assert parse(DoublePrecision, 'doubleprecision*4 a') == \
+        'DOUBLEPRECISION*4 a'
+    assert parse(DoublePrecision, 'double precision*4, a') == \
+        'DOUBLEPRECISION*4 a'
+    assert parse(DoublePrecision, 'doubleprecision*4 a ,b') == \
+        'DOUBLEPRECISION*4 a, b'
+    assert parse(DoublePrecision, 'double precision*4 :: a ,b') == \
+        'DOUBLEPRECISION*4 a, b'
+    assert parse(DoublePrecision, 'doubleprecision*4 a(1,2)') == \
+        'DOUBLEPRECISION*4 a(1,2)'
+    assert parse(DoublePrecision, 'double precision*4 :: a(1,2),b') == \
+        'DOUBLEPRECISION*4 a(1,2), b'
+    assert parse(DoublePrecision, 'doubleprecision*4 external :: a') == \
+        'DOUBLEPRECISION*4, external :: a'
+    assert parse(DoublePrecision, 'double precision*4, external :: a') == \
+        'DOUBLEPRECISION*4, external :: a'
+    assert parse(DoublePrecision,
+                 'double precision*4 external , intent(in) :: a') == \
+        'DOUBLEPRECISION*4, external, intent(in) :: a'
+    assert parse(DoublePrecision, 'doubleprecision(kind=4)') == \
+        'DOUBLEPRECISION(KIND=4)'
+    assert parse(DoublePrecision, 'double precision ( kind = 4)') == \
+        'DOUBLEPRECISION(KIND=4)'
+    assert parse(DoublePrecision, 'doubleprecision(kind=2+2)') == \
+        'DOUBLEPRECISION(KIND=2+2)'
+    assert parse(DoublePrecision, 'double precision(kind=f(4,5))') == \
+        'DOUBLEPRECISION(KIND=f(4,5))'
+
+
+def test_complex():
+    '''Test various declarations of complex variables. '''
+
+    assert parse(Complex, 'complex') == 'COMPLEX'
+    assert parse(Complex, 'complex*4') == 'COMPLEX*4'
+    assert parse(Complex, 'complex*4 a') == 'COMPLEX*4 a'
+    assert parse(Complex, 'complex*4, a') == 'COMPLEX*4 a'
+    assert parse(Complex, 'complex*4 a ,b') == 'COMPLEX*4 a, b'
+    assert parse(Complex, 'complex*4 :: a ,b') == 'COMPLEX*4 a, b'
+    assert parse(Complex, 'complex*4 a(1,2)') == 'COMPLEX*4 a(1,2)'
+    assert parse(Complex, 'complex*4 :: a(1,2),b') == \
+        'COMPLEX*4 a(1,2), b'
+    assert parse(Complex, 'complex*4 external :: a') == \
+        'COMPLEX*4, external :: a'
+    assert parse(Complex, 'complex*4, external :: a') == \
+        'COMPLEX*4, external :: a'
+    assert parse(Complex, 'complex*4 external , intent(in) :: a') == \
+        'COMPLEX*4, external, intent(in) :: a'
+    assert parse(Complex, 'complex(kind=4)') == 'COMPLEX(KIND=4)'
+    assert parse(Complex, 'complex ( kind = 4)') == 'COMPLEX(KIND=4)'
+    assert parse(Complex, 'complex(kind=2+2)') == 'COMPLEX(KIND=2+2)'
+    assert parse(Complex, 'complex(kind=f(4,5))') == \
+        'COMPLEX(KIND=f(4,5))'
+
+
+def test_double_complex():
+    '''Test various declarations of double complex variables. '''
+
+    assert parse(DoubleComplex, 'double complex') == 'DOUBLECOMPLEX'
+    assert parse(DoubleComplex, 'double complex*4') == 'DOUBLECOMPLEX*4'
```

Comment:
Again, my reading about this indicates that it is not valid to specify a kind for declarations of this type. (It appears to be a Fortran77 extension and thus pre-dates the introduction of KIND.)